src/compiler/crystal/macros/methods.cr

```diff
@@ -1533,6 +1538,10 @@ module Crystal
       ArrayLiteral.new(defs)
     end
 
+    def self.has_method?(type, name)
+      BoolLiteral.new(!!type.has_def?(name))
```

Comment:
I think that `BoolLiteral.new(type.has_def?(name))` is enough, as `Type#has_def?` is already returning a boolean